Add social_notifications_subscribe option in WP Admin

htdat opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

Fixes #709

When /wp-admin/options-discussion.php is fully loaded, the option looks like this: 

![Screen Shot on 2019-12-15 at 21:22:50](https://user-images.githubusercontent.com/10045087/70864067-39409f00-1f81-11ea-89e0-0a60cf1cabfe.png)
 
#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

Basically, I followed the same approach for another similar option ` Someone likes one of my posts` in "Likes" module https://github.com/Automattic/jetpack/search?q=social_notifications_like&type=Code

- Handle `social_notifications_subscribe` option when "Subscriptions" module is activated/deactivated.
- Add `Someone follows my blog` in WP Admin -> Settings -> Discussions.
- Use jQuery to align up this option in the `Email me whenever` section.
- Validate its value to only on/off.
- Add this option to \Jetpack_Core_Json_Api_Endpoints class. 

I do not add tests as `social_notifications_subscribe` is covered already. Can a dev help to double check this?

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?

* This is more an enhancement. See p9o2xV-3Q-p2 and https://github.com/Automattic/jetpack/issues/709#issuecomment-381045292

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Disable Subscriptions module => the option should not display in "Discussions" settings 
* Enable Subscriptions => the checkbox of this option should not be selected by default.
* Enable this option, try to subscribe with a new email => the Jetpack connection owner gets an email notification about the new email. 

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->

* Add `Email me whenever: Someone follows my blog` option to wp-admin settings.

htdat committed:
Add the social_notifications_subscribe option to wp-admin/options-discussion.php

zinigor reviewed (changes requested):
This change works well for me, but I have several comments. I don't really like how we're copying the same code that's used in the Likes module, but I guess that's the quickest way to fix the issue that we're aiming at.
Thanks for working on this!

htdat committed:
[not verified] Add coding style improvements and clarify a string for translators

htdat commented on modules/subscriptions.php:
```diff
@@ -371,6 +399,63 @@ function subscription_comment_subscribe_setting() {
+
+		<label>
+			<input type="checkbox" name="social_notifications_subscribe" id="social_notifications_subscribe" value="1" <?php checked( $checked ); ?> />
+			<?php _e( 'Someone follows my blog	', 'jetpack' ); ?>
```
> Why is there a tab symbol inside the translatable string after blog? If the label needs additional markup, let's move it outside of the i18n?

This was my mistake, my next commit will remove this tab symbol.

> Also this string really needs a translator comment, otherwise it's not clear what this string is for. Let's add something like:

Gotcha! This makes sense.

htdat commented:
@zinigor Thanks for your thorough review. I've added a new commit to address your suggestions. 

>  I don't really like how we're copying the same code that's used in the Likes module, but I guess that's the quickest way to fix the issue that we're aiming at.

Well, me too, I do not like this approach much. But since we have moved new options to Calypso or Jetpack setting pages, it's not efficient to refactor related code.

zinigor committed:
Added changes that were skipped in the previous commit.

zinigor committed:
Placed a comment iside a PHP block.

zinigor reviewed (approved):
Thanks for fixing everything! I have added the changes that linter required, please don't skip checks when you commit.
